program/src/state.rs

```diff
@@ -673,6 +673,32 @@ impl Default for Validator {
     }
 }
 
+impl Validator {
+    pub fn has_no_stake_accounts(&self) -> bool {
+        self.stake_seeds.begin == self.stake_seeds.end
+    }
+
+    pub fn check_can_be_removed(&self) -> Result<(), LidoError> {
+        if self.active {
+            msg!("Refusing to remove validator because it is still active, deactivate it first.");
+            return Err(LidoError::ValidatorIsStillActive);
+        }
+        if !self.has_no_stake_accounts() {
```

Comment:
We can simplify the double negation by having a method `has_stake_accounts` that checks `begin != end`.